designs/shelly/src/base.mjs

```diff
@@ -0,0 +1,210 @@
+  const hips = measurements.hips * (1 + hipsEase)
+
+  store.set('neckRadius', neckRadius)
+
+  points.raglanCenter = new Point(0, 0)
+  points.neckCenter = points.raglanCenter.shift(270, options.neckBalance * neckRadius)
+
+  points.armholeCorner = new Point(chest / 4, armholeYPosition)
+
+  points.neckShoulderCorner = utils.beamIntersectsCircle(
+    points.neckCenter,
+    neckRadius,
+    points.raglanCenter,
+    points.armholeCorner
+  )[1]
+
+  points.cfHem = new Point(0, bodyLength)
+  points.sideHem = new Point(hips / 4, bodyLength)
+  points.cfNeck = points.neckCenter.shift(270, neckRadius)
+
+  const raglanAngle = points.neckShoulderCorner.angle(points.armholeCorner)
+  store.set('raglanAngle', raglanAngle)
+  const raglanLength = points.raglanCenter.dist(points.armholeCorner)
+  store.set('raglanLength', raglanLength)
+
+  const necklineAngleAtRaglan = points.cfNeck.angle(points.neckShoulderCorner) * 2
+  const necklineArcLength = utils.deg2rad(necklineAngleAtRaglan) * neckRadius
+  points.neckCP1 = points.neckShoulderCorner.shift(
+    necklineAngleAtRaglan + 180,
+    necklineArcLength / 3
+  )
+  points.neckCP2 = points.cfNeck.shift(0, necklineArcLength / 3)
+
+  const frontNecklineToRaglanAngle = raglanAngle - (necklineAngleAtRaglan + 180)
+  store.set('frontNecklineToRaglanAngle', frontNecklineToRaglanAngle)
+
+  points.armholeCornerScooped = points.armholeCorner.shift(
+    raglanAngle + 90,
+    options.raglanScoopMagnitude * raglanLength
+  )
+  points.armholeScoopCp1 = points.armholeCorner.shift(
+    raglanAngle + 180,
+    (1 / 3) * options.raglanScoopLength * raglanLength
+  )
+  points.armholeScoopCp2 = points.armholeCorner.shift(
+    raglanAngle + 180,
+    (2 / 3) * options.raglanScoopLength * raglanLength
+  )
+  points.armholeScoopEnd = points.armholeCorner.shift(
+    raglanAngle + 180,
+    options.raglanScoopLength * raglanLength
+  )
+
+  // Make the side seams vertical if we're making a tubular shirt. Overrides any hips measurements or options.
+  if (options.straightSides)
+    points.sideHem.x = Math.max(points.armholeCornerScooped.x, points.sideHem.x)
+
+  // Make sure that the shirt at least reaches the armholes, to ensure that the full raglan seam can be formed. This code should only trigger is someone tries to make a really, _really_ short shirt.
+  if (points.sideHem.y < points.armholeCornerScooped.y) {
+    points.sideHem.y = points.armholeCornerScooped.y
+    points.cfHem.y = points.armholeCornerScooped.y
+  }
+
+  const sideAngle = points.sideHem.angle(points.armholeCornerScooped)
+  const sideLength = points.sideHem.dist(points.armholeCornerScooped)
+  points.sideCp1 = points.sideHem
+    .shift(sideAngle, (1 / 3) * sideLength)
+    .shift(sideAngle - 90, options.sideShape * sideLength)
+  points.sideCp2 = points.armholeCornerScooped
+    .shift(sideAngle + 180, (1 / 3) * sideLength)
+    .shift(sideAngle - 90, options.sideShape * sideLength)
+
+  const drawSide = () => {
```

Comment:
Same remark: Avoid defining functions in functions

Reply:
Fixed